subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -464,8 +465,26 @@ static int __wifi_args_to_params(size_t argc, char *argv[],
 			return -EINVAL;
 		}
 
-		if (params->channel == 0U) {
+		if (iface_mode == WIFI_MODE_INFRA && params->channel == 0) {
```

Comment:
Yes, I don't see any chipset supporting that feature, so, for now allowing only valid channels,if we implement ACS, then it would be much more involved than just allowing channel = 0.